From: David Henningsson <david.henningsson@canonical.com>
Date: Mon, 28 Mar 2011 08:19:16 +0000 (+0200)
Subject: Conf: Fix channels of ICE1712 to 10/12
X-Git-Url: http://git.alsa-project.org/?p=alsa-lib.git;a=commitdiff_plain;h=d3906a93072171e5b5f4000d4a228af4eb8fa253

Conf: Fix channels of ICE1712 to 10/12

The ICE1712 kernel driver accepts 10 playback channels and
12 recording channels only. Therefore fix this channel count
in alsa-lib.

BugLink: http://bugs.launchpad.net/bugs/178442
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---

diff --git a/src/conf/cards/ICE1712.conf b/src/conf/cards/ICE1712.conf
index 01e50d2..f5b8add 100644
--- a/src/conf/cards/ICE1712.conf
+++ b/src/conf/cards/ICE1712.conf
@@ -32,12 +32,26 @@ ICE1712.pcm.front.0 {
 	@args.CARD {
 		type string
 	}
-	type route
-	ttable.0.0 1
-	ttable.1.1 1
-	slave.pcm {
-		type hw
-		card $CARD
+	type asym
+	playback.pcm {
+		type route
+		ttable.0.0 1
+		ttable.1.1 1
+		slave.pcm {
+			type hw
+			card $CARD
+		}
+		slave.channels 10
+	}
+	capture.pcm {
+		type route
+		ttable.0.0 1
+		ttable.1.1 1
+		slave.pcm {
+			type hw
+			card $CARD
+		}
+		slave.channels 12
 	}
 }	
 
@@ -57,6 +71,7 @@ ICE1712.pcm.surround40.0 {
 		type hw
 		card $CARD
 	}
+	slave.channels 10
 }	
 
 <confdir:pcm/surround41.conf>
@@ -79,6 +94,7 @@ ICE1712.pcm.surround51.0 {
 		type hw
 		card $CARD
 	}
+	slave.channels 10
 }
 
 <confdir:pcm/iec958.conf>
